api/v1alpha1/nnfcontainerprofile_webhook_test.go

```diff
@@ -304,4 +309,28 @@ var _ = Describe("NnfContainerProfile Webhook", func() {
 		newProfile.Data.Pinned = false
 		Expect(k8sClient.Update(context.TODO(), newProfile)).ToNot(Succeed())
 	})
+
+	DescribeTable("when modes are set for storages on creation",
+		func(storageName string, mode corev1.PersistentVolumeAccessMode, result bool) {
+			for i, storage := range nnfProfile.Data.Storages {
+				if storage.Name == storageName && mode != "" {
+					nnfProfile.Data.Storages[i].PVCMode = mode
+				}
+			}
+			if result {
+				Expect(k8sClient.Create(context.TODO(), nnfProfile)).To(Succeed())
+			} else {
+				Expect(k8sClient.Create(context.TODO(), nnfProfile)).ToNot(Succeed())
+				nnfProfile = nil
+			}
+		},
+		// Only nil modes should pass for JOB/PERSISTENT
+		Entry("should pass when DW_JOB has no mode", "DW_JOB_storage", nil, true),
```

Comment:
PersistentVolumeAccessMode is a string, not a string pointer.  You could redefine 'mode' as `*corev1.PersistentVolumeAccessmode`, and adjust the conditional for that.  Or you could send an empty string.

I think it's confusing when you send a nil for something that's not a pointer.